File: tests/fixtures/repomap/sample.php
<?php

namespace App\Services;

use App\Models\User;
use App\Contracts\Repository;

class UserService
{
    public function getUser(int $id): User { }
    private function validate(int $id): bool { }
    protected function logAccess(): void { }
}

interface UserRepository
{
    public function find(int $id): ?User;
    public function save(User $user): void;
}

trait Cacheable
{
    public function getCacheKey(): string { }
}

enum UserRole: string
{
    case Admin = 'admin';
    case User = 'user';
}

function createApp(array $config): void { }

File: tests/fixtures/repomap/sample.cs
using System;
using System.Collections.Generic;

namespace App.Services
{
    public class UserService
    {
        public void GetUser(int id) { }
        private void Validate(int id) { }
        protected void LogAccess() { }
    }

    public interface IRepository
    {
        void Save(object entity);
    }

    public struct UserData
    {
        public int Id;
        public string Name;
    }

    public enum UserRole
    {
        Admin,
        User,
        Guest
    }
}
